src/script/client/ClientEntity.ts

```diff
@@ -53,62 +62,56 @@ class ClientEntity {
       isVerified: ko.observable(false),
       primaryKey: undefined,
     };
-
-    this.session = {};
   }
 
   /**
    * Splits an ID into user ID & client ID.
-   * @param {string} id - Client ID to be dismantled
-   * @returns {Object} Object containing the user ID & client ID
    */
-  static dismantleUserClientId(id) {
-    const [userId, clientId] = _.isString(id) ? id.split('@') : [];
+  static dismantleUserClientId(id: string): {clientId: string; userId: string} {
+    const [userId, clientId] = isString(id) ? id.split('@') : ([] as string[]);
     return {clientId, userId};
   }
 
   /**
    * Returns the ID of a client in a format suitable for UI display in user preferences.
-   * @returns {Array<string>} Client ID in pairs of two as an array
+   * @returns Client ID in pairs of two as an array
    */
-  formatId() {
+  formatId(): string[] {
     return zeroPadding(this.id, 16).match(/.{1,2}/g);
   }
 
   /**
-   * @returns {boolean} True, if the client is the self user's permanent client.
+   * @returns `true`, if the client is the self user's permanent client.
```

Comment:
There is no check for `isSelfClient`, so it will also be `true` if it is the client of someone else.

Reply:
This is a refactor and I don't think I want to introduce behaviour changes here. Maybe we can do it in a second PR?